package cl.model;

import java.time.LocalDate;

/**
 * Representa el estado y trazabilidad de un libro en circulación.
 * Registra y gestiona estado de entrega, plazo de devolución y fechas relevantes.
 */
public class Prestamo {
	private int id;
	private int idEstudiante;
	private int idLibro;
	private LocalDate fechaPrestamo;
	private LocalDate fechaDevolucion;
	private boolean devuelto;

	public Prestamo() {
		this.fechaPrestamo = LocalDate.now();
		this.fechaDevolucion = LocalDate.now().plusDays(7);
		this.devuelto = false;
	}

	public Prestamo(int id, int idEstudiante, int idLibro, LocalDate fechaPrestamo, LocalDate fechaDevolucion, boolean devuelto) {
		this.id = id;
		this.idEstudiante = idEstudiante;
		this.idLibro = idLibro;
		this.fechaPrestamo = fechaPrestamo;
		this.fechaDevolucion = fechaDevolucion;
		this.devuelto = devuelto;
	}

	public Prestamo(int idEstudiante, int idLibro, LocalDate fechaPrestamo, LocalDate fechaDevolucion, boolean devuelto) {
		this.idEstudiante = idEstudiante;
		this.idLibro = idLibro;
		this.fechaPrestamo = fechaPrestamo;
		this.fechaDevolucion = fechaDevolucion;
		this.devuelto = devuelto;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getIdEstudiante() {
		return idEstudiante;
	}

	public void setIdEstudiante(int idEstudiante) {
		this.idEstudiante = idEstudiante;
	}

	public int getIdLibro() {
		return idLibro;
	}

	public void setIdLibro(int idLibro) {
		this.idLibro = idLibro;
	}

	public LocalDate getFechaPrestamo() {
		return fechaPrestamo;
	}

	public void setFechaPrestamo(LocalDate fechaPrestamo) {
		this.fechaPrestamo = fechaPrestamo;
	}

	public LocalDate getFechaDevolucion() {
		return fechaDevolucion;
	}

	public void setFechaDevolucion(LocalDate fechaDevolucion) {
		this.fechaDevolucion = fechaDevolucion;
	}

	public boolean isDevuelto() {
		return devuelto;
	}

	public void setDevuelto(boolean devuelto) {
		this.devuelto = devuelto;
	}

	@Override
	public String toString() {
		return "Préstamo " + id + (devuelto ? " Devuelto" : "Pendiente");
	}
}
